roles/static_inventory/tasks/openstack.yml

```diff
@@ -0,0 +1,51 @@
+---
+- no_log: true
+  block:
+    - name: fetch all nodes from openstack shade dynamic inventory
+      command: shade-inventory --list
+      register: registered_nodes_output
+      when: refresh_inventory|bool
+
+    - name: set fact for openstack inventory cluster nodes
+      set_fact:
+        registered_nodes: "{{ (registered_nodes_output.stdout | from_json) | json_query(q) }}"
+      vars:
+        q: "[] | [?metadata.clusterid=='{{stack_name}}']"
+      when:
+        - refresh_inventory|bool
+
+    - name: set_fact for openstack inventory nodes
+      set_fact:
+        registered_nodes_floating: "{{ (registered_nodes_output.stdout | from_json) | json_query(q2) }}"
+      vars:
+        q: "[] | [?metadata.group=='infra.{{stack_name}}']"
+        q2: "[] | [?metadata.clusterid=='{{stack_name}}'] | [?public_v4!='']"
+      when:
+        - refresh_inventory|bool
+
+    - name: Add cluster nodes w/o floating IPs to inventory
+      with_items: "{{ registered_nodes }}"
+      when: not item in registered_nodes_floating
+      add_host:
+        name: '{{ item.name }}'
+        groups: '{{ item.metadata.group }}'
+        ansible_host: '{{ item.name }}'
+        ansible_fqdn: '{{ item.name }}'
+        ansible_user: '{{ ssh_user }}'
+        ansible_private_key_file: '{{ private_ssh_key }}'
+        ansible_ssh_extra_args: '-F {{ ssh_config_path }}'
+        private_v4: '{{ item.private_v4 }}'
+
+    - name: Add cluster nodes with floating IPs to inventory
+      with_items: "{{ registered_nodes_floating }}"
+      when: item in registered_nodes_floating
+      add_host:
+        name: '{{ item.name }}'
+        groups: '{{ item.metadata.group }}'
+        ansible_host: '{{ item.name }}'
```

Comment:
The bastion node should be able to access the private IPs (otherwise what's the point of it) and Ansible still prints out the item.name when running. From the point of view of the user running the playbooks nothing changes.